### PR TITLE
W2-hiepphan-Create calculator view

### PR DESCRIPTION
**Pull request by ghost**
#### Related issues (Optional)
Task 2

#### Description (Required)
- [x] Create Layout Calculator
- [x] Handling Algorithms of Calculator

### CODE REVIEW

File: app/src/at-hiepphan/res/layout/activity_calculator_view.xml [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+            android:background="@drawable/shape_border" />
+    </LinearLayout>
+
+    <TableLayout
+        android:layout_width="match_parent"
+        android:layout_height="wrap_content"
+        android:orientation="horizontal"
+        android:padding="5dp">
+
+        <TableRow>
```
Comment (at-dev-support-01) [inline app/src/at-hiepphan/res/layout/activity_calculator_view.xml]: WARNING: [UselessParent] This `TableRow` layout or its `TableLayout` parent is useless
 A layout with children that has no siblings, is not a scrollview or a root layout, and does not have a background, can be removed and have its children moved directly into the parent for a flatter and more efficient layout hierarchy.

File: app/src/at-hiepphan/res/layout/activity_calculator_view.xml [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+            <Button
```
Comment (at-dev-support-01) [inline app/src/at-hiepphan/res/layout/activity_calculator_view.xml]: WARNING: [ButtonStyle] Buttons in button bars should be borderless; use `style="?android:attr/buttonBarButtonStyle"` (and `?android:attr/buttonBarStyle` on the parent)
 Button bars typically use a borderless style for the buttons. Set the `style="?android:attr/buttonBarButtonStyle"` attribute on each of the buttons, and set `style="?android:attr/buttonBarStyle"` on the parent layout

File: app/src/at-hiepphan/res/layout/activity_calculator_view.xml [hunk 3]
```diff
@@ -0,0 +1,95 @@
...
+                android:id="@+id/btnAddition"
+                android:layout_height="wrap_content"
+                android:text="@string/textview.text.addition"
+                android:layout_weight="1"/>
+
+            <Button
```
Comment (at-dev-support-01) [inline app/src/at-hiepphan/res/layout/activity_calculator_view.xml]: WARNING: [ButtonStyle] Buttons in button bars should be borderless; use `style="?android:attr/buttonBarButtonStyle"` (and `?android:attr/buttonBarStyle` on the parent)
 Button bars typically use a borderless style for the buttons. Set the `style="?android:attr/buttonBarButtonStyle"` attribute on each of the buttons, and set `style="?android:attr/buttonBarStyle"` on the parent layout

File: app/src/at-hiepphan/res/layout/activity_calculator_view.xml [hunk 4]
```diff
@@ -0,0 +1,95 @@
...
+                android:id="@+id/btnSubtraction"
+                android:layout_height="wrap_content"
+                android:layout_weight="1"
+                android:text="@string/textview.text.subtraction" />
+
+            <Button
```
Comment (at-dev-support-01) [inline app/src/at-hiepphan/res/layout/activity_calculator_view.xml]: WARNING: [ButtonStyle] Buttons in button bars should be borderless; use `style="?android:attr/buttonBarButtonStyle"` (and `?android:attr/buttonBarStyle` on the parent)
 Button bars typically use a borderless style for the buttons. Set the `style="?android:attr/buttonBarButtonStyle"` attribute on each of the buttons, and set `style="?android:attr/buttonBarStyle"` on the parent layout

File: app/src/at-hiepphan/res/layout/activity_calculator_view.xml [hunk 5]
```diff
@@ -0,0 +1,95 @@
...
+                android:id="@+id/btnMultiplication"
+                android:layout_height="wrap_content"
+                android:text="@string/textview.text.multiplication"
+                android:layout_weight="1" />
+
+            <Button
```
Comment (at-dev-support-01) [inline app/src/at-hiepphan/res/layout/activity_calculator_view.xml]: WARNING: [ButtonStyle] Buttons in button bars should be borderless; use `style="?android:attr/buttonBarButtonStyle"` (and `?android:attr/buttonBarStyle` on the parent)
 Button bars typically use a borderless style for the buttons. Set the `style="?android:attr/buttonBarButtonStyle"` attribute on each of the buttons, and set `style="?android:attr/buttonBarStyle"` on the parent layout

**Comment by at-hiepphan** [comment]
@at-huongnguyen 
please review for me
Thank you

File: app/src/at-hiepphan/res/layout/activity_calculator_view.xml [hunk 6]
```diff
@@ -0,0 +1,98 @@
...
+    </LinearLayout>
+
+    <LinearLayout
+        android:weightSum="4"
+        android:layout_width="match_parent"
+        android:layout_height="wrap_content"
+        android:orientation="horizontal"
+        android:padding="5dp">
+
+        <Button
```
Comment (at-dev-support-01) [inline app/src/at-hiepphan/res/layout/activity_calculator_view.xml]: WARNING: [ButtonStyle] Buttons in button bars should be borderless; use `style="?android:attr/buttonBarButtonStyle"` (and `?android:attr/buttonBarStyle` on the parent)
 Button bars typically use a borderless style for the buttons. Set the `style="?android:attr/buttonBarButtonStyle"` attribute on each of the buttons, and set `style="?android:attr/buttonBarStyle"` on the parent layout

File: app/src/at-hiepphan/res/layout/activity_calculator_view.xml [hunk 7]
```diff
@@ -0,0 +1,98 @@
...
+            android:id="@+id/btnAddition"
+            android:layout_width="0dp"
+            android:layout_height="wrap_content"
+            android:layout_weight="1"
+            android:text="@string/textview.text.addition" />
+
+        <Button
```
Comment (at-dev-support-01) [inline app/src/at-hiepphan/res/layout/activity_calculator_view.xml]: WARNING: [ButtonStyle] Buttons in button bars should be borderless; use `style="?android:attr/buttonBarButtonStyle"` (and `?android:attr/buttonBarStyle` on the parent)
 Button bars typically use a borderless style for the buttons. Set the `style="?android:attr/buttonBarButtonStyle"` attribute on each of the buttons, and set `style="?android:attr/buttonBarStyle"` on the parent layout

File: app/src/at-hiepphan/res/layout/activity_calculator_view.xml [hunk 8]
```diff
@@ -0,0 +1,98 @@
...
+            android:id="@+id/btnSubtraction"
+            android:layout_width="0dp"
+            android:layout_height="wrap_content"
+            android:layout_weight="1"
+            android:text="@string/textview.text.subtraction" />
+
+        <Button
```
Comment (at-dev-support-01) [inline app/src/at-hiepphan/res/layout/activity_calculator_view.xml]: WARNING: [ButtonStyle] Buttons in button bars should be borderless; use `style="?android:attr/buttonBarButtonStyle"` (and `?android:attr/buttonBarStyle` on the parent)
 Button bars typically use a borderless style for the buttons. Set the `style="?android:attr/buttonBarButtonStyle"` attribute on each of the buttons, and set `style="?android:attr/buttonBarStyle"` on the parent layout

File: app/src/at-hiepphan/res/layout/activity_calculator_view.xml [hunk 9]
```diff
@@ -0,0 +1,98 @@
...
+            android:id="@+id/btnMultiplication"
+            android:layout_width="0dp"
+            android:layout_height="wrap_content"
+            android:layout_weight="1"
+            android:text="@string/textview.text.multiplication" />
+
+        <Button
```
Comment (at-dev-support-01) [inline app/src/at-hiepphan/res/layout/activity_calculator_view.xml]: WARNING: [ButtonStyle] Buttons in button bars should be borderless; use `style="?android:attr/buttonBarButtonStyle"` (and `?android:attr/buttonBarStyle` on the parent)
 Button bars typically use a borderless style for the buttons. Set the `style="?android:attr/buttonBarButtonStyle"` attribute on each of the buttons, and set `style="?android:attr/buttonBarStyle"` on the parent layout

File: app/src/at-hiepphan/java/vn/asiantech/internship/CalculatorView.java [hunk 10]
```diff
@@ -0,0 +1,165 @@
...
+    private void subtraction(BigDecimal firstNumber, BigDecimal secondNumber) {
+        mTvResult.setText(String.valueOf(firstNumber.subtract(secondNumber)));
+    }
+
+    private void multiplication(BigDecimal firstNumber, BigDecimal secondNumber) {
+        mTvResult.setText(String.valueOf(firstNumber.multiply(secondNumber)));
+    }
+
+    private void division(BigDecimal firstNumber, BigDecimal secondNumber) {
+        if (secondNumber.compareTo(new BigDecimal(0)) == 0) {
```
Comment (at-dev-support-01) [inline app/src/at-hiepphan/java/vn/asiantech/internship/CalculatorView.java]: WARNING: [BigIntegerInstantiation] Dont create instances of already existing BigInteger and BigDecimal (ZERO, ONE, TEN)
https://pmd.github.io/pmd-5.6.1/pmd-java/rules/java/basic.html#BigIntegerInstantiation

**Comment by at-hiepphan** [comment]
@at-huongnguyen
please review for me
Thank you

File: app/src/at-hiepphan/java/vn/asiantech/internship/CalculatorView.java [hunk 11]
```diff
@@ -0,0 +1,163 @@
...
+    @Override
+    protected void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+        setContentView(R.layout.activity_calculator_view);
+        initView();
+        onClickListener();
+        checkInputData();
+    }
+
+    private void initView() {
```
Comment (at-huongnguyen) [inline app/src/at-hiepphan/java/vn/asiantech/internship/CalculatorView.java]: `initViews`

File: app/src/at-hiepphan/java/vn/asiantech/internship/CalculatorView.java [hunk 12]
```diff
@@ -0,0 +1,163 @@
...
+        mEdtSecondNumber = findViewById(R.id.edtSecondNumber);
+        mBtnAddition = findViewById(R.id.btnAddition);
+        mBtnSubtraction = findViewById(R.id.btnSubtraction);
+        mBtnMultiplication = findViewById(R.id.btnMultiplication);
+        mBtnDivision = findViewById(R.id.btnDivision);
+        mTvOperator = findViewById(R.id.tvOperator);
+        mTvResult = findViewById(R.id.tvResult);
+    }
+
+    private void onClickListener() {
```
Comment (at-huongnguyen) [inline app/src/at-hiepphan/java/vn/asiantech/internship/CalculatorView.java]: Should rename to `addListener()`

File: app/src/at-hiepphan/java/vn/asiantech/internship/CalculatorView.java [hunk 13]
```diff
@@ -0,0 +1,163 @@
...
+                    break;
+                case R.id.btnDivision:
+                    mTvOperator.setText(this.getString(R.string.textview_text_division));
+                    division(firstNumber, secondNumber);
+                    break;
+            }
+        }
+    }
+
+    private void checkInputData() {
```
Comment (at-huongnguyen) [inline app/src/at-hiepphan/java/vn/asiantech/internship/CalculatorView.java]: Should rename to `addTextChangedListener()`

**Pull request closed without merging**